File: backend/cart-service/src/main/java/com/constams/cartservice/model/Product.java
package com.constams.cartservice.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
public class Product {
    private String name;
    private String description;
    private BigDecimal price;
    private String skuCode;
}

File: frontend/src/app/model/Product.ts
export class Product {
  name: string;
  description: string;
  price: number;
  skuCode: string;

  constructor(
    name: string,
    description: string,
    price: number,
    skuCode: string
  ) {
    this.name = name;
    this.description = description;
    this.price = price;
    this.skuCode = skuCode;
  }
}
